### DISCUSSION TITLE
Got the "Print" function to recognise my custom template, how I force the renderer to use that template too?

### DISCUSSION THREAD

**Discussion by fyrfli** [kimai/kimai#2302] [Questions & Answers]
Hi.

After much futzing, I managed to create my custom template in var/export using the default as a starting point. When I hit "Print" on the export interface, I am able to select that custom template and it displays as intended. Now I want to get the "PDF" option to use that custom template as well and I seem to be hitting a wall.

I copied the export/renderer folder and altered the default.pdf.twig to have a different name as well as reference the var/export/custom-template.pdf.twig (extends var/export/custom-pdf-layout.html.twig); but it seems to still be using the default.pdf.twig from template/export/renderer.

What am I missing? Can anyone help?

ETA: Can I tag on a request about creating a grand total for the summary table as well?

**Comment by kevinpapst** [comment]
Sorry, I am sure it makes sense for you, but for me it doesn't.
Please post your new directory structure and filenames: which new files are located where, starting from the root of the kimai installation.

Does the new export template show up in the PDF dropdown, next to the 3 default ones?

> ETA: Can I tag on a request about creating a grand total for the summary table as well?

Which summary table, what grand total?

**Reply by fyrfli** [reply]
I'm sorry. Let me post the directory structure with files:


/$kimai-homedir/var
   |_     cache
           data
           easy_backup
           export
             |_    aloe-pdf-layout.html.twig
                    renderer
                        |_    aloe-default-budget.pdf.twig
                               aloe-default.html.twig
                               aloe-default-internal.pdf.twig
                               aloe-default-pdf.twig
        invoices
        log
        packages
        plugins
        sessions
vendor

(The spaces are not showing up. Trying again.)

I hope this is clearer. And the PDF dropdown does *not* show any other option than the 3 defaults ones.

**Reply by fyrfli** [reply]
![image](https://user-images.githubusercontent.com/5287261/106696343-208ab280-65a2-11eb-8310-3776825b7470.png)

**Reply by kevinpapst** [reply]
move everything from var/export/renderer to var/export/ and (maybe) flush the cache ... and please read the docs, thats what they are written for: https://www.kimai.org/documentation/export.html#adding-export-templates

**Reply by fyrfli** [reply]
😊 Yes. I tried really hard to RTFM, but there were some details that seem obscure to me. It might be just my ignorance, which is why I reached out to the community (at least I thought I did). For example, there is the reference to config keys, but I couldn't find where those config keys should reside. Also, the documentation says "put your templates in var/export since that directory doesn't ship with updates" but it doesn't say whether we need to keep the same directory structure (hence me creating a sub-directory called "renderer" like it is in the templates/export directory). 

I moved everything out of renderer and the pdf now has dropdown options for internal and budget, but not the "standard". I tried different naming schemes, ended up with aloe-pdf.twig, reloaded the environment, but I still don't see an option for that one. The content of aloe-pdf.twig is 

`
{% extends 'var/export/aloe-pdf-layout.html.twig' %}
`

I feel like I may be missing something foundational in understanding all this.

**Reply by kevinpapst** [reply]
> HTML templates have the file extension .html.twig
> PDF templates have the file extension .pdf.twig

Yours `-pdf.twig`

**Reply by fyrfli** [reply]
Which is where I am confused because mine-pdf.twig is not showing as a dropdown option; mine.pdf.twig does, however, and gives me a critical error.

![image](https://user-images.githubusercontent.com/5287261/106936826-ac0b5d00-66e2-11eb-836c-4d88cce42dff.png)

So I am clearly missing something.

**Reply by kevinpapst** [reply]
So your original problem is solved: the template shows up in the dropdown if you use a correct filename?!

A screenshot of a generic error is obviously not saying aynthing 😁 
You should switch to developer mode (`APP_ENV=dev` and `composer install` afterwards) to see proper stack traces.
You can try to solve your problem by checking logs at Doctor> Logs (or var/logs/prod.log) as well.

**Reply by fyrfli** [reply]
Yeah. Original issue solved. But, as with anything to do with software, often brings more problems. :D

And my next step was indeed to start scouring logs. 

I'm gonna close this now. I can always re-open or open a new issue if I am running into further issues.

Again, thanks for you time and patience.

ETA: Oh ... I can't close.
